frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -123,6 +134,27 @@ private boolean isBranchNameRepeated(String fileContent, String branchName) {
     return lines.stream().filter(line -> line.equals(branchName)).count() > 1;
   }
 
+  @SuppressWarnings("allcheckers")
+  private String getParentBranchName(String fileContent, String branchName) {
```

Comment:
Huh I'd reeeealy consider using `BranchLayoutReader` instead of custom logic 😅

Reply:
Ok, done :)